certbot/tests/main_test.py

```diff
@@ -1404,6 +1404,44 @@ def test_update_account_no_existing_accounts(self):
                  "user@example.org"])
             self.assertTrue("Could not find an existing account" in x[0])
 
+    @mock.patch('certbot._internal.main._determine_account')
+    @mock.patch('certbot._internal.eff.prepare_subscription')
+    @mock.patch('certbot._internal.main.account')
+    def test_update_account_remove_email(self, mocked_account_module, mock_prepare, mock_det_acc):
+        # Mock account storage and the account object returned
+        mocked_storage = mock.MagicMock()
+        mocked_account = mock.MagicMock()
+
+        mocked_account_module.AccountFileStorage.return_value = mocked_storage
+        mocked_storage.find_all.return_value = [mocked_account]
+        mock_det_acc.return_value = (mocked_account, "foo")
+
+        # Mock client and mock registration body to verify calls are made
+        mock_regr_body = mock.MagicMock()
+
+        # mocked_account.regr is overwritten in update, requiring an odd mock setup
+        mocked_account.regr.body = mock_regr_body
+
+        x = self._call(
+            ["update_account", "--register-unsafely-without-email"])
+
+
+        # When update succeeds, the return value of update_account() is None
+        self.assertTrue(x[0] is None)
+        # and we got supposedly did update the registration from
+        # the server
+        client_mock = x[3]
+
+        self.assertEqual(client_mock.mock_calls[1][0], 'Client().acme.update_registration')
```

Comment:
I think this is relying on the exact order calls were made on the client object which seems fragile to me. Instead, what about something like:
```
        mock_update_reg = client_mock.Client().acme.update_registration
        self.assertTrue(mock_update_reg.called)
```

Reply:
Ah thank you! The python documentation on mocks elides practical usage and I did not realize I could include the `Client()` in the call chain. That's much better and I appreciate the suggestion.